src/main/kotlin/org/web3j/container/ContainerBuilder.kt

```diff
@@ -37,10 +42,15 @@ class ContainerBuilder {
         this.genesisPath = genesisPath
     }
 
-    fun build(): KGenericContainer {
+    fun withSelfAddress(selfAddress: String) = apply {
+        this.selfAddress = selfAddress
+    }
+
+    fun build(): IKGenericContainer {
         return when (type) {
             NodeType.BESU -> BesuContainer(version, resourceFiles, hostFiles, genesisPath)
             NodeType.GETH -> GethContainer(version, resourceFiles, hostFiles, genesisPath)
+            NodeType.LOCAL -> LocalContainer(Configuration(Address(selfAddress), 10), PassthroughTracer())
```

Comment:
I think this should go in a separate builder

Reply:
Or maybe have an abstract service builder and separate container service builder and local service builder

Reply:
Not sure about this, let's think again after the first round of changes.